src/Feature.cpp

```diff
@@ -132,4 +132,282 @@ const std::vector<Feature*>& Feature::GetFeatureList()
+	}
+}
+
+template <typename T>
+void Feature::RenderImGuiElement(const std::string& settingName, const gameSetting& settingData, T* valuePtr)
+{
+	// Unique ID for each element
+	ImGui::PushID(settingName.c_str());  // Use settingName as unique ID
+
+	if constexpr (std::is_same_v<T, bool>) {
+		ImGui::Checkbox(settingData.friendlyName.c_str(), valuePtr);
+	} else if constexpr (std::is_same_v<T, float>) {
+		try {
+			auto minFloat = std::get<float>(settingData.minValue);
+			auto maxFloat = std::get<float>(settingData.maxValue);
+			ImGui::SliderFloat(settingData.friendlyName.c_str(), valuePtr, minFloat, maxFloat);
+		} catch (const std::bad_variant_access&) {
+			logger::warn("Type mismatch for {}: expected float for minValue or maxValue but received other type", settingName);
+		}
+	} else if constexpr (std::is_same_v<T, int> || std::is_same_v<T, unsigned int>) {
+		try {
+			auto minInt = std::get<int32_t>(settingData.minValue);
+			auto maxInt = std::get<int32_t>(settingData.maxValue);
+			ImGui::SliderInt(settingData.friendlyName.c_str(), reinterpret_cast<int*>(valuePtr), minInt, maxInt);
+		} catch (const std::bad_variant_access&) {
+			logger::warn("Type mismatch for {}: expected int for minValue or maxValue but received other type", settingName);
+		}
+	} else {
+		logger::warn("Unsupported type for {}: {}", settingName, typeid(T).name());
+	}
+
+	// Tooltip handling
+	if (auto _tt = Util::HoverTooltipWrapper()) {
+		std::string hover = "";
+		if (settingData.offset != 0)
+			hover = std::format("{}\n\nNOTE: CS cannot save this game setting directly. Game setting '{}' might be able to be saved manually in the ini. Use the Copy button to export to clipboard.", settingData.description, settingName);
+		else
+			hover = settingData.description;
+		ImGui::Text(hover.c_str());
+	}
+	if (settingData.offset != 0) {
+		ImGui::SameLine();
+		if (ImGui::Button("Copy")) {
+			ImGui::SetClipboardText(settingName.c_str());
+		}
+		if (auto _tt = Util::HoverTooltipWrapper()) {
+			ImGui::Text(std::format("Copy '{}' to clipboard.", settingName).c_str());
+		}
+	}
+	ImGui::PopID();  // End unique ID scope
+}
+
+void Feature::RenderImGuiElement(const std::string& settingName, const gameSetting& settingData, RE::Setting* setting)
+{
+	// Determine the type of the setting
+	switch (setting->GetType()) {
+	case RE::Setting::Type::kBool:
+		RenderImGuiElement(settingName, settingData, &setting->data.b);
+		break;
+	case RE::Setting::Type::kFloat:
+		RenderImGuiElement(settingName, settingData, &setting->data.f);
+		break;
+	case RE::Setting::Type::kSignedInteger:
+		RenderImGuiElement(settingName, settingData, &setting->data.i);
+		break;
+	case RE::Setting::Type::kUnsignedInteger:
+		RenderImGuiElement(settingName, settingData, &setting->data.u);
+		break;
+	case RE::Setting::Type::kColor:
+	case RE::Setting::Type::kString:
+	case RE::Setting::Type::kUnknown:
+	default:
+		logger::warn("Unsupported type for setting '{}'", settingName);
```

Comment:
Why no color or string?

Reply:
Are we really going to be exposing those as a setting? I only exposed what I needed for dynamic cubemaps. I could see an argument for color if someone really wanted it, but no idea why we'd expose a string.